pepsico/monthly_seasonal_conv_mapping.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+import plotly.express as px
+from dash import Dash, dcc, html, Input, Output
+from urllib.request import urlopen
+import json
+
+def compute_annual_monthly_avg(ds, variable, start_year=None, end_year=None):
+    #compute the avg (variable) for each month across the specified years
+    
+    #if precipitation variable, change from kg to mm per day
+    if variable == 'pr':
+        ds[variable] *= 86400
+    
+    if start_year and end_year:
+        ds = ds.sel(T=slice(f"{start_year}-01-01", f"{end_year}-12-31"))
+
+    # Convert daily data to monthly
+    monthly_ds = ds.resample(T="1M").mean()
+
+    # Compute the average for each month across all years
+    monthly_avg = monthly_ds.mean(dim=["X", "Y"])
+    annual_monthly_avg = monthly_avg.groupby("T.month").mean(dim="T")
+    df = annual_monthly_avg.to_dataframe().reset_index()
+    return df
+
+def compute_annual_seasonal_avg(monthly_ds, variable, start_year=None, end_year=None):
+    #compute monthly avgs across all years
+    monthly_avg = monthly_ds.mean(dim=["X", "Y"])
+
+    # Compute rolling seasonal average (3-month rolling window, center-aligned)
+    rolling_avg = monthly_avg.rolling(T=3, center=True).mean()
+    
+    # Compute the average variable for each rolling season across all years
+    rolling_seasonal_avg = rolling_avg.groupby("T.month").mean(dim="T")
+    
+    # Create a DataFrame for seasonal averages
+    seasonal_avgs_df = rolling_seasonal_avg.to_dataframe().reset_index()
+
+    # Assign the season column with correct length
+    seasonal_avgs_df['season'] = ['DJF', 'JFM', 'FMA', 'MAM', 'AMJ', 'MJJ',
+                                   'JJA', 'JAS', 'ASO', 'SON', 'OND', 'NDJ']
+
+    return seasonal_avgs_df
+    
+
+def write_to_csv(df, scenario, model, variable, output_dir):
+
+    # Write to CSV file
+    file_path = f'{output_dir}/{scenario}_{model}_{variable}_annual_monthly_avg.csv'
+    df.to_csv(file_path, index=False)
+
+
+def map_averages(df, model, variable):
+
+    df = df.sort_values(by='month')
+    map_fig = px.imshow(
+        df.pivot(index='Y', columns='X', values=variable), 
```

Comment:
both compute_annual_ functions are averaging over X and Y so they don't exist anymore. Are you trying to make maps? or time series? or both?